web/src/components/Header.js

```diff
@@ -1,42 +1,110 @@
 
+import Icon, { faChevronDown, faChevronLeft, faEdit, faSignOutAlt } from './Icons';
+
 class Header extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { ariaExpanded: props.ariaExpanded };
+    this.handleOutsideClick = this.handleOutsideClick.bind(this);
+  }
+
+  componentDidMount() {
+    document.addEventListener('mousedown', this.handleOutsideClick);
+  }
+
+  handleOutsideClick = e => {
+    if (this.node.contains(e.target)) {
+      return;
+    }
+    this.setState({ ariaExpanded: false })
+  };
+
   handleLogout = e => {
     e.preventDefault();
     const { pushRoute } = this.props;
     pushRoute('/logout');
   };
 
+  toggleDropdown = () => {
+    this.setState(prev => ({ ariaExpanded: !prev.ariaExpanded }));
+  };
+
   render() {
-    const { authenticated } = this.props;
+    const { authenticated, currentUser, isAdmin, location } = this.props;
+    const { ariaExpanded } = this.state;
+    const isTopLevel = location.pathname === '/';
     return (
-      <header className="header__slim">
+      <header ref={node => this.node = node}>
         <div className="ds-l-container">
           <div className="ds-l-row">
-            <div className="ds-l-col--12 ds-l-md-col--4">
-              <Link to="/" className="site-title">
-                {t('titleBasic')}
-              </Link>
+            <div className="ds-l-col--12 ds-l-md-col--4 site-title">
+              {isTopLevel &&
+                <Link to="/">
+                  {t('titleBasic')}
+                </Link>
+              }
+              {!isTopLevel && !authenticated &&
+                <Link to="/">
+                  {t('titleBasic')}
+                </Link>
+              }
+              {!isTopLevel && authenticated && (
+                <Link to="/">
+                  <Icon icon={faChevronLeft} size="sm" />
+                  { isAdmin ? "Admin dashboard" : `${currentUser.state.id.toUpperCase()} APD home` }
+                </Link>
+              )}
             </div>
             {authenticated &&
-              <div className="ds-l-col--12 ds-l-md-col--4 ds-u-margin-left--auto">
-                <Button
-                  size="small"
-                  className="action--logout"
-                  variation="transparent"
-                  onClick={this.handleLogout}
-                >
-                  {t('logout')}
-                </Button>
+              <div className="ds-l-col--12 ds-l-md-col--8">
+                <ul className="nav--dropdown" aria-expanded={ariaExpanded}>
+                  <li>
+                    <Button
+                      size="small"
+                      variation="transparent"
+                      className="nav--dropdown__trigger"
+                      onClick={this.toggleDropdown}
+                    >
+                      { currentUser ? currentUser.username : 'Your account' }
+                      <Icon icon={faChevronDown} style={{ width: '8px'}} />
```

Comment:
Not a huge fan of this pattern, but it's the only way I can find to reliably set the SVG width. Setting it in the CSS and not inline on the element seems to get ignored. In summary, SVGs are black magic.

Reply:
I did try that! But for some reason, the icons were all slightly different sizes and it looked _weird_. ![no idea](https://media1.giphy.com/media/bPTXcJiIzzWz6/giphy.gif)